pennylane/tape/interfaces/jax.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the mixin interface class for creating differentiable quantum tapes with
+JAX.
+"""
+from functools import partial
+import jax
+import jax.experimental.host_callback as host_callback
+import jax.numpy as jnp
+from pennylane.tape.queuing import AnnotatedQueue
+from pennylane.operation import Variance, Expectation
+
+
+class JAXInterface(AnnotatedQueue):
+    """Mixin class for applying an JAX interface to a :class:`~.JacobianTape`.
+
+    JAX-compatible quantum tape classes can be created via subclassing:
+
+    .. code-block:: python
+
+        class MyAutogradQuantumTape(AutogradInterface, JacobianTape):

```

Comment:
```suggestion
        class MyJAXQuantumTape(JAXInterface, JacobianTape):
```

Reply:
P.S. Let's move the interfaces to the device ASAP before the interfaces grow even more 😆

Reply:
Agreed.